controllers/front/product.php

```diff
@@ -28,36 +28,42 @@ class ShoppingfeedProductModuleFrontController extends \ModuleFrontController
 {
     protected $sfToken = null;
 
-    public function initContent()
+    public function init()
     {
-        $token = Tools::getValue('token');
-        if (empty($token)) {
-            $token = (new ShoppingfeedToken())->getDefaultToken();
-        } else {
-            $token = (new ShoppingfeedToken())->findByToken($token);
-        }
+        parent::init();
+
+        $this->sfToken = (new ShoppingfeedToken())->findByToken(Tools::getValue('token', ''));
+    }
 
-        if ($token === false) {
-            exit();
+    public function checkAccess()
+    {
+        if (empty($this->sfToken)) {
+            header('HTTP/1.1 401 Unauthorized');
```

Comment:
Could you please use https://www.php.net/manual/en/function.http-response-code.php instead ? It’s compatible with http 1.1 or http 2.0 according to the merchand server

Reply:
Done